src/css-loader-core/loader.js

```diff
@@ -65,43 +66,61 @@ export default class FileSystemLoader {
 	fetch(_newPath, relativeTo, _trace) {
 		let newPath = _newPath.replace(/^["']|["']$/g, ""),
 			trace = _trace || String.fromCharCode(this.importNr++);
+		const useFileResolve = typeof this.fileResolve === "function";
 		return new Promise((resolve, reject) => {
-			let relativeDir = path.dirname(relativeTo),
-				rootRelativePath = path.resolve(relativeDir, newPath),
-				fileRelativePath = path.resolve(
-					path.resolve(this.root, relativeDir),
-					newPath
-				);
+			(useFileResolve
```

Comment:
Let's rewrite this using async/await. I'm thinking of rewriting all the promises into async/await in the future